iphone/TitaniumKit/TitaniumKit/Sources/API/TiWindowProxy.m

```diff
@@ -759,6 +759,9 @@ - (void)viewWillAppear:(BOOL)animated
   if (hidesBackButton) {
     [self setHidesBackButton:hidesBackButton];
   }
+  if (isModal && opening) {
+    [self windowDidOpen];
+  }
```

Comment:
I've tested this an it works. However, moving the `windowDidOpen()` call here causes the `Ti.UI.Window` object's "open" event to be fired earlier than it used to. I'm not sure if this has negative consequences or not, but an alternative solution would be to revert your code change here and instead change the [TiUINavigationWindowProxy.m](https://github.com/appcelerator/titanium_mobile/blob/master/iphone/Classes/TiUINavigationWindowProxy.m) code to set up the navigation controller via the `viewWillAppear()` method, which is what your code change does. I've tested the below and it works for me.
```objc
// Delete this old code since it doesn't set up nav controller in time.
#if 0
- (void)windowDidOpen
{
  // Set parent for navigation controller
  if (navController) {
    UIViewController *parentController = [self windowHoldingController];
    [parentController addChildViewController:navController];
    [navController didMoveToParentViewController:parentController];
  }
  [super windowDidOpen];
}

- (void)viewWillAppear:(BOOL)animated
{
  // Move the above code here.
  if (navController) {
    UIViewController *parentController = [self windowHoldingController];
    [parentController addChildViewController:navController];
    [navController didMoveToParentViewController:parentController];
  }

  if ([self viewAttached]) {
    [navController viewWillAppear:animated];
  }
  [super viewWillAppear:animated];
}
#endif
```

Reply:
Or we can move that block of code to the [TiUINavigationWindowProxy.m](https://github.com/appcelerator/titanium_mobile/blob/master/iphone/Classes/TiUINavigationWindowProxy.m) class' `windowWillOpen()` method. I've tested this and it work as well. The key thing is the nav controller needs to be set up before the window is opened.

Reply:
Fine with both! Natively, the `open` equivalent is `viewDidAppear`, so that should be kept as-is.